core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/stopstartus/StopStartUpscaleActions.java

```diff
@@ -89,12 +89,33 @@ protected void prepareExecution(StopStartUpscaleTriggerEvent payload, Map<Object
 
             @Override
             protected void doExecute(StopStartUpscaleContext context, StopStartUpscaleTriggerEvent payload, Map<Object, Object> variables) throws Exception {
-                clusterUpscaleFlowService.startingInstances(context.getStack().getId(), payload.getHostGroup(), payload.getAdjustment());
-                sendEvent(context);
+                clusterUpscaleFlowService.initScaleUp(context.getStack().getId(), context.getHostGroupName());
+                if (payload.isFailureRecoveryEnabled()) {
```

Comment:
Same comment as downscale: Nit: Put the following lookups inside the if condition, and generate empty lists otherwise.